pyfa_tool/dataset.py

```diff
@@ -0,0 +1,797 @@
+
+
+        fields_jsonpath = os.path.join(tmpdir, 'fields.json')
+        metadata_jsonpath = os.path.join(tmpdir, 'metadata.json')
+
+        # Read the json files
+        fielddata = IO.read_json(jsonpath=fields_jsonpath,
+                                 to_dataframe=True)
+        metadata = IO.read_json(jsonpath=metadata_jsonpath,
+                                 to_dataframe=False)
+
+        # Remove trailing and leading whitespace from fieldnames
+        fielddata['name'] = [fieldname.strip() for fieldname in fielddata['name']]
+
+        # update attributes
+        self.metadata = metadata
+        self.fielddf = fielddata
+
+        IO.remove_tempdir(tmpdir)
+
+
+
+
+class FaDataset():
+    """A Class that holds data, and methods, of an FA file."""
+
+    def __init__(self, fafile=None, nodata=-999):
+        """
+        Initiate of an FaDataset object.
+
+        Parameters
+        ----------
+        fafile : str, optional
+            Path of the target FA-file. The default is None.
+        nodata : int, optional
+            The Nodata value to be used by (rio)xarray. The default is -999.
+
+        Returns
+        -------
+        None.
+
+        Note
+        -------
+        Rasterio uses an integer value of the nodata, so make sure an
+        integer-casting of the nodata value exists.
+
+        """
+        # test if file exist
+        if not fafile is None:
+            if not IO.check_file_exist(fafile):
+                sys.exit(f'{fafile} is not a file.')
+
+        self.fafile = fafile
+        self.ds = None # xarray.Dataset
+        self.nodata = nodata
+
+
+    # =========================================================================
+    # Specials
+    # =========================================================================
+    def __repr__(self):
+        return str(self.ds)
+
+    def __str__(self):
+        return str(self.ds)
+
+
+    # =========================================================================
+    # Setters / Getters
+    # =========================================================================
+    def set_fafile(self, fafile):
+        """
+        Update the path to the FA-file.
+
+        Parameters
+        ----------
+        fafile : str
+            Path to the FA-file.
+
+        Returns
+        -------
+        None.
+
+        """
+        if not IO.check_file_exist(fafile):
+            sys.exit(f'{fafile} is not a file.')
+        self.fafile = fafile
+
+    def get_fieldnames(self):
+        """
+        Get all present variables
+
+        Returns
+        -------
+        list
+            A list of all the availabel variables.
+
+        """
+        return self._get_physical_variables()
+
+
+    # =========================================================================
+    # Importing data
+    # =========================================================================
+    def import_fa(self, whitelist=None, blacklist=None,
```

Comment:
Whitelist (and blacklist) currently for 3D fields currently only work with Base-names. So selecting a specific model level of for example snow is currently not possible. This may be a nice enhancement, if you agree I will make a new issue (but i think it is not a requirement for this version) as it may not be so simple.

Reply:
Converted in an issue #40